shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java

```diff
@@ -233,16 +235,19 @@ private String buildRpcExt(final ServiceBean<?> serviceBean, final String method
                 .url("")
                 .serialization(serviceBean.getSerialization())
                 .build();
-        // set method config: loadbalance,retries,timeout,sent
-        Optional.ofNullable(serviceBean.getMethods()).orElse(Collections.emptyList()).stream()
-                .filter(m -> methodName.equals(m.getName()))
-                .findFirst()
-                .ifPresent(methodConfig -> {
-                    Optional.ofNullable(methodConfig.getLoadbalance()).filter(StringUtils::isNotEmpty).ifPresent(build::setLoadbalance);
-                    Optional.ofNullable(methodConfig.getRetries()).ifPresent(build::setRetries);
-                    Optional.ofNullable(methodConfig.getTimeout()).ifPresent(build::setTimeout);
-                    Optional.ofNullable(methodConfig.getSent()).ifPresent(build::setSent);
-                });
+        // method config: loadbalance,retries,timeout,sent
+        if (Objects.nonNull(serviceBean.getMethods())) {
+            build.setMethods(new ArrayList<>());
+            for (MethodConfig methodConfig : serviceBean.getMethods()) {
+                DubboRpcMethodExt methodExt = new DubboRpcMethodExt();
+                methodExt.setName(methodConfig.getName());
+                methodExt.setLoadbalance(methodConfig.getLoadbalance());
```

Comment:
is not be null？

Reply:
The configuration of the method inherits from the interface. If it is null, the configuration of the interface will be used. So setting default values when null is not a good idea